test/e2e_node/container_restart_test.go

```diff
@@ -0,0 +1,121 @@
+package e2enode
+
+import (
+	"context"
+	"time"
+
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	kubeletconfig "k8s.io/kubernetes/pkg/kubelet/apis/config"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/kubernetes/test/e2e/feature"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+const containerName = "restarts"
+
+var _ = SIGDescribe("Container Restart", feature.CriProxy, framework.WithSerial(), func() {
+	f := framework.NewDefaultFramework("container-restart")
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+
+	ginkgo.Context("Container restart backs off", func() {
+
+		ginkgo.BeforeEach(func() {
+			if err := resetCRIProxyInjector(e2eCriProxy); err != nil {
+				ginkgo.Skip("Skip the test since the CRI Proxy is undefined.")
+			}
+		})
+
+		ginkgo.AfterEach(func() {
+			err := resetCRIProxyInjector(e2eCriProxy)
+			framework.ExpectNoError(err)
+		})
+
+		ginkgo.It("Container restart backs off.", func(ctx context.Context) {
+			// 0s, 0s, 10s, 30s, 70s, 150s, 310s
+			doTest(ctx, f, 5, containerName, 7)
+		})
+	})
+
+	ginkgo.Context("Alternate container restart backs off as expected", func() {
+
+		tempSetCurrentKubeletConfig(f, func(ctx context.Context, initialConfig *kubeletconfig.KubeletConfiguration) {
+			initialConfig.CrashLoopBackOff.MaxContainerRestartPeriod = &metav1.Duration{Duration: time.Duration(30 * time.Second)}
+			initialConfig.FeatureGates = map[string]bool{"KubeletCrashLoopBackOffMax": true}
+		})
+
+		ginkgo.BeforeEach(func() {
+			if err := resetCRIProxyInjector(e2eCriProxy); err != nil {
+				ginkgo.Skip("Skip the test since the CRI Proxy is undefined.")
+			}
+		})
+
+		ginkgo.AfterEach(func() {
+			err := resetCRIProxyInjector(e2eCriProxy)
```

Comment:
This file is not build-tag-limited to Linux-only, but is referencing variables like this which are defined in other files which are build tagged to be Linux only

Reply:
I think we can just tag this test as linux-only.

Reply:
Thanks! Fixed in [e6059d7](https://github.com/kubernetes/kubernetes/pull/128559/commits/e6059d7386479770ab5588dd33cc168b5c76e2e2)